route/access_rules.go

```diff
@@ -0,0 +1,128 @@
+package route
+
+import (
+	"errors"
+	"fmt"
+	"log"
+	"net"
+	"net/http"
+	"strings"
+)
+
+const (
+	ipAllowTag = "allow:ip"
+	ipDenyTag  = "deny:ip"
+)
+
+// AccessDeniedHTTP checks rules on the target for HTTP proxy routes.
+func (t *Target) AccessDeniedHTTP(r *http.Request) bool {
+	host, _, err := net.SplitHostPort(r.RemoteAddr)
+
+	if err != nil {
+		log.Printf("[ERROR] Failed to get host from RemoteAddr %s: %s",
+			r.RemoteAddr, err.Error())
+		return false
+	}
+
+	// prefer xff header if set
```

Comment:
I have some mixed feelings about this section.  In the case where fabio may be behind another proxy you would want to honor `XFF` but then what about the actual originator?  Would it be better to check both the `XFF` (if present and different) as well as the actual source and return `true` from this function if either the `XFF` or `RemoteAddr` pass returns `true`?